### ISSUE TITLE
[Question] No completion list when cursor after "(" because score=0

### ISSUE THREAD

**Issue by vatosarmat** [hrsh7th/nvim-cmp#1355]
Hello, and thanks you for amazing plugin.

I have a small but sometimes annoying issue with completion for `luasnip`. If I `cmp.complete` when cursor is at the beginning of the line or right after space, then popup with snippets list appears as expected. But if I do it when cursor is inside brackets, no popup appears. I found out that this is due to the fact that `score=0` for each completion item. Why it is 0? Is there anything I can do to see the completion list without necessity to enter `space`? 

Also, could you please explain what `cmp.ContextReason` does and what it influence? I din't manage to find it in documentation. Thanks.

Cursor is right after space - list appears:
![image](https://user-images.githubusercontent.com/35213999/208317388-77a31ac2-aca7-4627-b4ab-d3d604b42d31.png)

Cursor is right after `(` - nothing appears because `score=0`:
![image](https://user-images.githubusercontent.com/35213999/208317493-dac3c380-0e9d-44e9-a23b-420422070793.png)

Here is my completion mapping. Autocompletion is disabled, I trigger it manually.
```lua
  vim.keymap.set('i', '<F10>', function()
    cmp.complete({
      reason = cmp.ContextReason.Manual,
      config = {
        sources = {
          {
            name = 'luasnip'
          }
        }
      }
    })
  end)

```

**Comment by Shougo** [comment]
I think it is expected behavior.
Because your `setInterval(` text may be snippet trigger.
`nvim-cmp` does not know what is snippet trigger.  So it assumes non space text is snippet trigger.

**Comment by vatosarmat** [comment]
Yes, it thinks "(" is part of snippet trigger. Is this logic correct? I was thinking that snippet trigger should be alphanumeric.

**Comment by Shougo** [comment]
It seems difficult.  Because `setInterval(` snippet is useful if you want to complete the function arguments.

**Comment by hrsh7th** [comment]
It's caused by keyword pattern that defined by source side.

I met similar problem on cmp-vsnip source and I tried to solve this via https://github.com/hrsh7th/cmp-vsnip/commit/18ff44cd4fa873d8c2aa23cb26cdd68e326f9e9c .

But it isn't works as perfect yet.

If I mange this issue, I'll re-visit here.
